### DISCUSSION TITLE
Quick question - verbatim-like environments containing $

### DISCUSSION THREAD

**Discussion by ddbrierton** [James-Yu/LaTeX-Workshop#3920] [Q&A] [answered]
Hi,

I just came across issue #237 from 2017 which was closed as unable to fix. I was just wondering if there has been any change since then? It’s not a huge deal, but I am having the exact same problem with one document I’m working on.

Thanks in advance.

**Comment by jlelong** [comment]
Unfortunately, this is a limitation of vscode's syntax highlighting engine.

**Reply by ddbrierton** [reply]
Thank you for taking the time to reply, it’s much appreciated. I am a little puzzled by this limitation though, as there are no problems for syntax highlighting when the `verbatim` environment itself is used:

<img width="449" alt="verbatim" src="https://github.com/James-Yu/LaTeX-Workshop/assets/7255543/412ecd61-b479-49bd-b7f6-679e28586184">

as compared to when the environment is different:

<img width="449" alt="listing" src="https://github.com/James-Yu/LaTeX-Workshop/assets/7255543/28028911-c20e-409e-88c6-299148cc404a">

**Reply by jlelong** [reply, marked as answer]
The `lstlisting` example is a bug. This is fixed by https://github.com/jlelong/vscode-latex-basics/commit/bbeea0e4566bf7e482b132a346773b4da583c75c

Going back to the original question. The limitation is that a grammar cannot use configuration variables. Currently the following environments are considered as verbatim `[a-zA-Z]*code\*?` or `(fboxv|boxedv|V|v|spv)erbatim\*?` and all the environments defined by the `minted` package.

**Reply by ddbrierton** [reply]
Ah, now I understand completely. Thank you so much for taking the time to explain the issue.
